tests/support/test.tcl

```diff
@@ -86,7 +86,7 @@ proc assert_error {pattern code {detail ""}} {
     if {[catch {uplevel 1 $code} error]} {
         assert_match $pattern $error
     } else {
-        assert_failed "assertion:Expected an error but nothing was caught" $detail
+        assert_failed "Expected an error matching '$pattern' but got '$error'" $detail
```

Comment:
when we get here no exception was thrown (you can't log `$error`).
i suppose it's a good idea to remove the `assertion:` prefix, and add the `$pattern`.
p.s. maybe add `$detail` to the `assert_match` line above.

Reply:
why can't log `$error`, i have tried it and can indeed get the return
> If script does not raise an error, catch will return 0 (TCL_OK) and set the variable to the value returned from script.
```
Expected an error matching 'ERR*' but got 'OK' (context: type eval line 4 cmd {assert_error "ERR*" {r set a b}} proc ::test)
```

when we expect an exception but there is actually no exception, i think the return value is useful

Reply:
you're right. i forgot `catch` returns the result in no exception was thrown.
so your change looks good..
so let's just pass `$detail` to the assertion part too.